### PR TITLE
Allow F10 to propagate from inventory and support held hot keys

### PR DESCRIPTION
**Pull request by sicoderealm**
While the inventory was activate the F10 key was being consumed to activate the main game menu - we now allow this key to propagate and also support throttled held hot keys from the inventory as per the main viewport.

Fixes: #528

### CODE REVIEW

**Comment by Dirhai** [comment]
I tested this pull request and it worked perfectly, the bug with the F10 key has been resolved.

File: module/merintr/inventry.c [hunk 1]
```diff
@@ -796,7 +796,7 @@ Bool InventoryClick(HWND hwnd, BOOL fDoubleClick, int x, int y, UINT keyFlags)
 Bool InventoryKey(HWND hwnd, UINT key, Bool fDown, int cRepeat, UINT flags)
 {
    ID id;
-   Bool held_down = (flags & 0x4000) ? True : False;  /* Is key being held down? */
+   Bool held_down = (flags & 0x4000) ? true : false;  /* Is key being held down? */
```
Comment (akirmse) [inline module/merintr/inventry.c:799]: This change is incorrect.. the allowable values for Bool are True and False.  You can change it to bool if you want.

File: module/merintr/inventry.c [hunk 2]
```diff
@@ -835,25 +831,24 @@ Bool InventoryKey(HWND hwnd, UINT key, Bool fDown, int cRepeat, UINT flags)
       break;
 
    default:
-      inform = False;   // Modules will be informed in PerformAction call below
+      inform = false;   // Modules will be informed in PerformAction call below
```
Comment (akirmse) [inline module/merintr/inventry.c:834]: Incorrect change (next several instances too).

File: module/merintr/inventry.c [hunk 3]
```diff
@@ -473,6 +464,15 @@ LRESULT CALLBACK InventoryProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lP
 	  StatsDrawBorder();
       InventoryReleaseCapture();
       break;
+   case WM_SYSKEYDOWN:
+      // Prevent activation of the window's menu bar and propagate to parent
+      if (wParam == VK_F10)
+      {
+          if (HANDLE_WM_KEYDOWN_BLAK(hwnd, wParam, lParam, InventoryKey) == True)
+              return 0;
+          break;
```
Comment (akirmse) [inline module/merintr/inventry.c:473]: I don't think this break is needed.

File: clientd3d/intrface.c [hunk 4]
```diff
@@ -27,6 +27,8 @@ static Bool interface_created = False;  /* True iff interface has been created *
 extern font_3d				gFont;
 extern d3d_driver_profile	gD3DDriverProfile;
 
+static DWORD last_text_command;
```
Comment (akirmse) [inline clientd3d/intrface.c:30]: This name makes it sound like it's the last text command that was entered.  last_text_command_time, and add a comment?

File: clientd3d/intrface.c [hunk 5]
```diff
@@ -522,10 +474,17 @@ void PerformAction(int action, void *action_data)
       break;
 
    case A_TEXTCOMMAND:
-      TextInputSetText((char *) action_data, False);
-      ParseGotText((char *) action_data);
-      break;
-
+   {
+       DWORD now = timeGetTime();
```
Comment (akirmse) [inline clientd3d/intrface.c:478]: I'm not convinced this part is a good idea.

- It overrides the keyboard repeat rate from Windows system settings.
- Does this allow super fast macroing?

File: module/merintr/inventry.c [hunk 6]
```diff
@@ -809,10 +806,6 @@ Bool InventoryKey(HWND hwnd, UINT key, Bool fDown, int cRepeat, UINT flags)
    if (action == A_NOACTION)
       return False;
 
-   // Skip held-down keys, except for moving cursor around
```
Comment (akirmse) [inline module/merintr/inventry.c:812]: Presumably we were doing this for a reason?  What has changed?

**Pull request merged**